### PR TITLE
Spike: Add mixed number input

### PR DESCRIPTION
**Pull request by ntsummers1**
## Description
Sometimes the number you've reached just can't be gotten ahold of.

### How to test
./dev local
Create a report in the dashboard
Go to the test page
Play with the optional number input!

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/verbiage/errors.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export const schemaValidationErrors = {
   INVALID_URL: "Response must be a valid hyperlink/URL",
   INVALID_DATE: "Response must be a valid date",
   INVALID_END_DATE: "End date can't be before start date",
+  INVALID_PERCENTAGE: "Please input a Number OR 'N/A'",
```
Comment (braxex) [inline services/ui-src/src/verbiage/errors.ts:18]:
```suggestion
  INVALID_PERCENTAGE: "Response must be a valid number OR \"N/A\"",
```

File: services/ui-src/src/utils/forms/schemas.ts [hunk 2]
```diff
@@ -16,6 +16,21 @@ export const number = () =>
     .required(error.REQUIRED_GENERIC);
 export const numberOptional = () => number().notRequired();
 
+// Mixed
+export const numberOrNA = () =>
+  numberSchema()
+    .transform(validateNA)
+    .required()
+    .nullable()
+    .typeError("Please input a Number OR 'N/A'");
```
Comment (braxex) [inline services/ui-src/src/utils/forms/schemas.ts:25]:
```suggestion
    .typeError(error.INVALID_NUMBER_OR_NA);
```

File: services/ui-src/src/utils/forms/schemas.ts [hunk 3]
```diff
@@ -16,6 +16,21 @@ export const number = () =>
...
+
+function validateNA(value: any, originalValue: string) {
+  if (typeof originalValue === "string" && originalValue === "N/A") {
+    return null;
+  }
+  return value;
+}
```
Comment (braxex) [inline services/ui-src/src/utils/forms/schemas.ts:32]: Two thoughts:
  1. Not sure if we're going to be passing `null` here or `N/A`, but we'll figure this out.
  2. I think we might need to accept more than just `N/A` (for instance `N/A`, `n/a`, `Not applicable`, etc.). Maybe the `originalValue === "N/A"` check becomes a `.includes` check against an array of valid not-applicable-adjacent answers? Should prob build this way to future-proof.

**Review by gmrabian — CHANGES_REQUESTED** [review]
Help me out if I'm missing something:
I created my own number field and it allowed any text.
Screen Shot 2022-09-01 at 7 19 23 AM

**Review by gmrabian — APPROVED** [review]

**Review by braxex — APPROVED** [review]
Looks awesome. Only one tweak. 👍

File: services/ui-src/src/verbiage/errors.ts [hunk 4]
```diff
@@ -10,11 +10,11 @@ export const schemaValidationErrors = {
   REQUIRED_GENERIC: "A response is required",
   REQUIRED_CHECKBOX: "Select at least one response",
   INVALID_GENERIC: "Response must be valid",
-  INVALID_NUMBER: "Response must be a valid number",
   INVALID_EMAIL: "Response must be a valid email address",
   INVALID_URL: "Response must be a valid hyperlink/URL",
   INVALID_DATE: "Response must be a valid date",
   INVALID_END_DATE: "End date can't be before start date",
+  INVALID_NUMBER_OR_NA: 'Response must be a valid number OR "N/A"',
```
Comment (braxex) [inline services/ui-src/src/verbiage/errors.ts:17]:
```suggestion
  INVALID_NUMBER_OR_NA: 'Response must be a valid number or "N/A"',
```

**Pull request merged**